common/plugins/org.jboss.tools.common.ui/src/org/jboss/tools/common/ui/JBossPerspectiveFactory.java

```diff
@@ -68,35 +73,39 @@ public void createInitialLayout(IPageLayout layout) {
 
 		layout.addShowInPart(ProjectExplorer.VIEW_ID);
 
+		Set<String> knownViews = new HashSet<String>();
+
 		String editorArea = layout.getEditorArea();
 
 		// Top left.
 		IFolderLayout topLeft = layout.createFolder("topLeft", IPageLayout.LEFT, 0.25f, editorArea);//$NON-NLS-1$
-		topLeft.addView(ProjectExplorer.VIEW_ID);
-		topLeft.addView(JavaUI.ID_PACKAGES);
-		topLeft.addPlaceholder(IPageLayout.ID_RES_NAV);
-		topLeft.addPlaceholder(JavaUI.ID_TYPE_HIERARCHY);
-		topLeft.addPlaceholder(JavaUI.ID_PACKAGES_VIEW);
-		topLeft.addPlaceholder(TestRunnerViewPart.NAME);
+		topLeft.addView(add(ProjectExplorer.VIEW_ID, knownViews));
+		topLeft.addView(add(JavaUI.ID_PACKAGES, knownViews));
+		topLeft.addPlaceholder(add(IPageLayout.ID_RES_NAV, knownViews));
+		topLeft.addPlaceholder(add(JavaUI.ID_TYPE_HIERARCHY, knownViews));
+		topLeft.addPlaceholder(add(JavaUI.ID_PACKAGES_VIEW, knownViews));
+		topLeft.addPlaceholder(add(TestRunnerViewPart.NAME, knownViews));
 
 		// Bottom right.
 		IFolderLayout bottomRight = layout.createFolder("bottomRight", IPageLayout.BOTTOM, 0.7f, editorArea);//$NON-NLS-1$
-		bottomRight.addView(IPageLayout.ID_PROBLEM_VIEW);
+		bottomRight.addView(add(IPageLayout.ID_PROBLEM_VIEW, knownViews));
 //		bottomRight.addView(IPageLayout.ID_PROP_SHEET);
-		bottomRight.addView(ID_SERVERS_VIEW);
+		bottomRight.addView(add(ID_SERVERS_VIEW, knownViews));
 
-		IFolderLayout bottomRight2 = layout.createFolder("bottomRight2", IPageLayout.RIGHT , 0.7f, "bottomRight");//$NON-NLS-1$
-		bottomRight2.addView(IPageLayout.ID_PROP_SHEET);
+		IFolderLayout bottomRight2 = layout.createFolder("bottomRight2", IPageLayout.RIGHT , 0.6f, "bottomRight");//$NON-NLS-1$
+		bottomRight2.addView(add(IPageLayout.ID_PROP_SHEET, knownViews));
 
-		bottomRight.addPlaceholder(IPageLayout.ID_TASK_LIST);
-		bottomRight.addPlaceholder(ID_CONSOLE_VIEW);
-		bottomRight.addPlaceholder(IPageLayout.ID_BOOKMARKS);
-		bottomRight.addPlaceholder(IProgressConstants.PROGRESS_VIEW_ID);
-		bottomRight.addPlaceholder(ID_SEARCH_VIEW);
+		bottomRight.addPlaceholder(add(IPageLayout.ID_TASK_LIST, knownViews));
+		bottomRight.addPlaceholder(add(ID_CONSOLE_VIEW, knownViews));
+		bottomRight.addPlaceholder(add(IPageLayout.ID_BOOKMARKS, knownViews));
+		bottomRight.addPlaceholder(add(IProgressConstants.PROGRESS_VIEW_ID, knownViews));
+		bottomRight.addPlaceholder(add(ID_SEARCH_VIEW, knownViews));
 
 		// Top right.
 		IFolderLayout topRight = layout.createFolder("topRight", IPageLayout.RIGHT, 0.7f, editorArea);//$NON-NLS-1$
-		topRight.addView(IPageLayout.ID_OUTLINE);
+		topRight.addView(add(IPageLayout.ID_OUTLINE, knownViews));
+
+		assignAllUnknownViewsToDefaultFolder(bottomRight, knownViews);
```

Comment:
So, it will assign all unknown views to the bottom right? Isn't it the default eclipse behavior which you want to override?